transmission_interface/src/bidirectional_effort_joint_interface_provider.cpp

```diff
@@ -0,0 +1,87 @@
+
+// ros_control
+#include <transmission_interface/bidirectional_effort_joint_interface_provider.h>
+
+namespace transmission_interface
+{
+    bool BiDirectionalEffortJointInterfaceProvider::registerTransmission(TransmissionLoaderData &loader_data,
+            TransmissionHandleData &handle_data)
+    {
+        // Setup joint state interface first (if not yet done)
+        if(!hasResource(handle_data.name, loader_data.transmission_interfaces.act_to_jnt_state))
+        {
+            if(!JointStateInterfaceProvider::registerTransmission(loader_data, handle_data))
+            {
+                return false;
+            }
+        }
+
+        if(!hasResource(handle_data.name, loader_data.inverse_transmission_interfaces.jnt_to_act_state))
+        {
+            if(!loader_data.robot_transmissions->get<transmission_interface::JointToActuatorStateInterface>())
+            {
+                loader_data.robot_transmissions->registerInterface(&loader_data.inverse_transmission_interfaces.jnt_to_act_state);
+            }
+
+            transmission_interface::JointToActuatorStateInterface &interface = *(loader_data.robot_transmissions->get<transmission_interface::JointToActuatorStateInterface>());
+
+            // Update transmission interface
+            transmission_interface::JointToActuatorStateHandle handle(handle_data.name, handle_data.transmission.get(), handle_data.act_state_data, handle_data.jnt_state_data);
+            interface.registerHandle(handle);
+        }
+
+        // If command interface does not yet exist in the robot transmissions, add it and use internal data structures
+        if(!loader_data.robot_transmissions->get<transmission_interface::JointToActuatorEffortInterface>())
+        {
+            loader_data.robot_transmissions->registerInterface(&loader_data.transmission_interfaces.jnt_to_act_eff_cmd);
+        }
+
+        if(!loader_data.robot_transmissions->get<transmission_interface::ActuatorToJointEffortInterface>())
+        {
+            loader_data.robot_transmissions->registerInterface(&loader_data.inverse_transmission_interfaces.act_to_jnt_eff_cmd);
+        }
+
+        transmission_interface::JointToActuatorEffortInterface &interface = *(loader_data.robot_transmissions->get<transmission_interface::JointToActuatorEffortInterface>());
+        transmission_interface::ActuatorToJointEffortInterface &reverse_interface = *(loader_data.robot_transmissions->get<transmission_interface::ActuatorToJointEffortInterface>());
+
+        // Setup command interface
+        transmission_interface::JointToActuatorEffortHandle handle(handle_data.name, handle_data.transmission.get(), handle_data.act_cmd_data, handle_data.jnt_cmd_data);
+        interface.registerHandle(handle);
+
+        transmission_interface::ActuatorToJointEffortHandle reverse_handle = transmission_interface::ActuatorToJointEffortHandle(handle_data.name, handle_data.transmission.get(), handle_data.act_cmd_data, handle_data.jnt_cmd_data);
```

Comment:
This looks like greater than 120 columns, please follow the style in https://github.com/jlack1987/ros_control/blob/dfcb72375a3378b99e64971a46ef5f612f7dbbca/transmission_interface/src/effort_joint_interface_provider.cpp#L138